src/celeritas/optical/PreGenDistributionData.hh

```diff
@@ -36,14 +38,14 @@ struct OpticalStepData
 /*!
  * Input data for sampling optical photons.
  */
-struct OpticalDistributionData
+struct PreGenDistributionData
```

Comment:
Maybe `Dispatcher` instead of `PreGenerator`? In the sense that optical photons are dispatched to another part of the code (can't use off-loader or transporter). Can also try `Deferer`, `Forwarder`, `Memoizer`, etc.